docs/rfcs/0000-template.md

```diff
@@ -0,0 +1,46 @@
+---
+layout: page
+title: your title
+permalink: /docs/rfcs/file-name
+---
+
+* RFC PR: [PR #](https://github.com/mozilla-mobile/android-components/pull/#)
+* Start date: YYYY-MM-DD (Day of proposal posting.)
+* End date: YYYY-MM-DD (Last day for general feedback. However, the proposal can be merged immediately after all stakeholders approve.)
+* Stakeholders: github-username, github-username
+
+## Summary
+
+This section should include a brief description of the proposal.
+
+## Motivation
+
+This section should include reasoning about why the proposal is useful. Examples, specific scenarios, 
+open bugs, records of performance metrics, and other empirical data are beneficial to include here if available.
+
+## Guide-level explanation
+
+This section should include a high-level walkthrough of the steps required to implement the proposal.
+
+## Reference-level explanation (optional)
+
+This section should include a detailed walkthrough of technical steps or code changes that 
+will be required to implement the proposal. Code samples and prototypes are beneficial to include here.
+
+## Drawbacks (optional)
+
+This section should include any drawbacks to the proposal.
+
+## Rationale and alternatives (optional)
+
+This section should include any alternative proposals considered, as well as the rationale for why
+they were not selected for the proposal.
+
+## Resources and Docs (optional)
+
+- Any (internal or external) similar proposals  or other documentation that shares concepts with the proposal.
+- Links to artifacts generated as part of the proposal, such as additional documentation or follow-up bugs.
+
+## Unresolved questions (optional)
```

Comment:
```suggestion
## Unresolved questions
```

Let's not make this optional. If there are no unresolved questions, then this section would require the developer to be explicit that they considered it and couldn't find any. Otherwise, all optional sections will most likely always be ignored.